### ISSUE TITLE
Map re-center when i drag it or change zoom

### ISSUE THREAD

**Issue by nschurmann** [airbnb/react-native-maps#1567]
Hello, happens with the basic configuration, this is my component:

```javascript
import _ from 'lodash'
import React, { Component } from 'react'
import { View, ScrollView, Modal, Dimensions, Platform, TouchableOpacity } from 'react-native'
import { Content, List, ListItem, Text, Header, Button, Icon, Item, Input } from 'native-base'
import IconMaterial from 'react-native-vector-icons/MaterialIcons'
import IconEntypo from 'react-native-vector-icons/Entypo'
import MapView from 'react-native-maps'

import PlacesAutocomplete from '../../Components/placesAutocomplete'

import config from '../../../config'
const Spinner = require('react-native-spinkit')

const { height, width } = Dimensions.get('window')

const styles = {
  confirm: {
    position: 'absolute',
    top: height - 110,
    margin: 20,
    width: width - 40,
  },
  locationIcon: {
    color: '#fff',
    fontSize: 16,
    top: 6,
    backgroundColor: '#FF5722',
    margin: 0,
    marginRight: 10,
    padding: 7,
    borderRadius:15,
    overflow: 'hidden',
    height: 31,
  },
  streetName: {
    alignSelf: 'flex-start',
    fontSize: 18,
    color: '#414E5F',
    left: 2,
  },
  localityName: {
    alignSelf: 'flex-start',
    fontSize: 12,
    color: '#8492A6',
  },
  row: {
    flexDirection: 'row',
    width: '100%',
  }
}

const shortenString = str => {
  return str.split(',')[0].length > 30 ? `${str.split(',')[0].substr(0,27)}...` : str.split(',')[0]
}

class Fetching extends Component {
  render() {
    return (
      <ListItem style={{ justifyContent: 'center'}}>
        <Spinner isVisible={true} size={40} type={'Bounce'} color={'#03B0EE'} />
      </ListItem>
    )
  }
}

class NoResults extends Component {
  render() {
    return (
      <ListItem style={{ justifyContent: 'center' }}>
        <Text style={styles.streetName}>No se encontraron resultados</Text>
        <IconEntypo style={[styles.streetName, { top: 3, left: 5 }]} name='emoji-sad' />
      </ListItem>
    )
  }
}

class AddressItem extends Component {
  render() {
    const { description } =this.props

    return (
      <ListItem key={description} style={{ height: 55, paddingBottom: 17}}>
        <TouchableOpacity style={styles.row}>
          <IconMaterial style={styles.locationIcon} name='location-on' />
          <View>
            <Text style={styles.streetName}>
              {shortenString(description)}
            </Text>
            <Text style={styles.localityName}>{_.tail(description.split(',')).join(',')}</Text>
          </View>
        </TouchableOpacity>
      </ListItem>
    )
  }
}

export default class AddressModal extends Component {
  state = {
    fetching: false,
    showList: false,
    results: [],
  }

  getResults(results) {
    this.setState({ results });
  }

  handleChange ({ address, fetching }) {
    if(address === ''){
      return this.setState({ showList: false, fetching })
    }
    this.setState({ showList: true, fetching })
  }

  handleCloseButton() {
    this.props.hideModal()
    this.setState({ showList: false, results: [] })
  }

  render() {
    
    return (
      <Modal
        animationType={"slide"}
        transparent={false}
        visible={this.props.showModal}
        >
        <View>
          <Header
            searchBar
            rounded
            style={{ backgroundColor: '#fff', borderBottomColor: '#fff' }}
            iosBarStyle='dark-content'
            >
            <Item style={{ borderBottomColor: '#A7B5C6' }}>
              <PlacesAutocomplete
                change={::this.handleChange}
                apiKey={config.placesKey}
                getResults={::this.getResults} />
              <Icon name="ios-search" />
            </Item>
            <Button transparent onPress={::this.handleCloseButton}>
              <Icon style={{ color: '#A7B5C6' }} name="close" />
            </Button>
          </Header>
          <View>
          </View>
          <MapView
            style={{ height: Platform.OS === 'ios' ? height - 76 : 52, width }}
            region={{
              latitude: 37.78825,
              longitude: -122.4324,
              latitudeDelta: 0.0922,
              longitudeDelta: 0.0421,
            }}
          />
          {this.state.showList &&
            <ScrollView style={{ backgroundColor: '#fff',top: this.state.showList ? 62 - height : 0}}>
              <List>
                {this.state.fetching ?
                  <Fetching />
                  : (!this.state.fetching && this.state.results.length > 0)
                    ? this.state.results.map(x => <AddressItem key={x.description} {...x} />)
                    : <NoResults />}
              </List>
            </ScrollView>}
          <Button large style={styles.confirm} block><Text>CONFIRMAR</Text></Button>
        </View>
      </Modal>
    )
  }
}
```

Happens in version 0.16.0 and 0.16.1

**Comment by nschurmann** [comment]
Happens in 0.16.0 even with initialRegion, but fixed in 0.16.1 if i change it to initialRegion.

**Comment by joegoodall1** [comment]
I've got the same problem with 0.16.0 (and getting build errors after upgrading to 0.16.1).
